apps/src/blocksCommon.js

```diff
@@ -162,3 +163,33 @@ function installWhenRun(blockly, skin, isK1) {
     return '\n';
   };
 }
+
+function installJoinBlock(blockly) {
+  blockly.Blocks.text_join_simple = {
+    init: function () {
+      this.helpUrl = '';
+      this.setColour(160);
+      this.setOutput(true, Blockly.BlockValueType.STRING);
+      this.setTooltip(commonMsg.JoinTextTooltip());
+    },
+
+    setInputCount: function (inputCount) {
+      this.inputCount = parseInt(inputCount);
```

Comment:
if we instead have this function _only_ set input count (and also define it directly on Blockly.Block, as I suggested in the review of the Blockly PR), I believe we can have init reference `this.inputCount`, and attach all inputs in init where we traditionally expect them to be attached.